src/plugins/discover/public/application/main/components/layout/use_discover_histogram.ts

```diff
@@ -127,56 +172,109 @@ export const useDiscoverHistogram = ({
       isPlainRecord
         ? undefined
         : {
-            status: hitsFetchStatus,
+            status: hitsFetchStatus.toString() as UnifiedHistogramFetchStatus,
             total: hitsTotal,
           },
     [hitsFetchStatus, hitsTotal, isPlainRecord]
   );
 
-  const { fetchStatus: chartFetchStatus, response, error } = useDataState(savedSearchData$.charts$);
+  /**
+   * Chart
+   */
 
-  const { bucketInterval, chartData } = useMemo(
-    () =>
-      buildChartData({
-        data,
-        dataView,
-        timeInterval: state.interval,
-        response,
-      }),
-    [data, dataView, response, state.interval]
+  const onChartHiddenChange = useCallback(
+    (chartHidden: boolean) => {
+      // Clear the Lens request adapter when the chart is hidden
+      if (chartHidden) {
+        inspectorAdapters.lensRequests = undefined;
+      }
+
+      storage.set(CHART_HIDDEN_KEY, chartHidden);
+      stateContainer.setAppState({ hideChart: chartHidden });
+    },
+    [inspectorAdapters, stateContainer, storage]
+  );
+
+  const onChartLoad = useCallback(
+    (event: UnifiedHistogramChartLoadEvent) => {
+      // We need to store the Lens request adapter in order to inspect its requests
+      inspectorAdapters.lensRequests = event.adapters.requests;
+    },
+    [inspectorAdapters]
   );
 
   const chart = useMemo(
     () =>
       isPlainRecord || !isTimeBased
         ? undefined
         : {
-            status: chartFetchStatus,
             hidden: state.hideChart,
             timeInterval: state.interval,
-            bucketInterval,
-            data: chartData,
-            error,
           },
-    [
-      bucketInterval,
-      chartData,
-      chartFetchStatus,
-      error,
-      isPlainRecord,
-      isTimeBased,
-      state.hideChart,
-      state.interval,
-    ]
+    [isPlainRecord, isTimeBased, state.hideChart, state.interval]
+  );
+
+  /**
+   * Breakdown
+   */
+
+  const onBreakdownFieldChange = useCallback(
+    (breakdownField: DataViewField | undefined) => {
+      stateContainer.setAppState({ breakdownField: breakdownField?.name });
+    },
+    [stateContainer]
   );
 
+  const field = useMemo(
+    () => (state.breakdownField ? dataView.getFieldByName(state.breakdownField) : undefined),
+    [dataView, state.breakdownField]
+  );
+
+  const breakdown = useMemo(
+    () => (isPlainRecord || !isTimeBased || !field ? undefined : { field }),
+    [field, isPlainRecord, isTimeBased]
+  );
+
+  /**
+   * Reload
+   */
+
+  const [lastReloadRequestTime, setLastReloadRequestTime] = useState(0);
+  const { fetchStatus: mainFetchStatus } = useDataState(savedSearchData$.main$);
+  const firstRun = useRef(true);
+
+  // Reload unified histogram when a refetch is triggered,
+  // with a debounce to avoid multiple requests
+  const [, cancelDebounce] = useDebounce(
```

Comment:
Another way to handle this could be subscribing to `savedSearchData$`, set setLastReloadRequestTime when  `FetchStatus.LOADING` es emitted. also deboucing could be implemented with the observable like here:

https://github.com/elastic/kibana/blob/581d6f8592a2c44d524d5ed3afd4b0be254068eb/src/plugins/discover/public/application/main/utils/get_fetch_observable.ts#L62